src/libraries/Microsoft.PowerFx.Core/Public/Values/CollectionTableValue.cs

```diff
@@ -65,6 +73,20 @@ public override int Count()
             }
         }
 
+        public override async Task<DValue<RecordValue>> AppendAsync(RecordValue record)
+        {
+            if (_sourceList == null)
```

Comment:
We can also check ICollection<T>.IsReadOnly 

[https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.icollection-1.isreadonly?view=net-6.0](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.icollection-1.isreadonly?view=net-6.0)

If it's readonly, we should just call base class (and let it fail)

       if (_sourceList == null || _sourceList.IsReadOnly)

Reply:
Done.